lib/guess_html_encoding.rb

```diff
@@ -45,7 +45,7 @@ def self.encode(html, headers = nil)
     if html_copy.valid_encoding?
       html_copy
     else
-      html_copy.force_encoding('ASCII-8BIT').encode('UTF-8', :undef => :replace, :invalid => :replace)
```

Comment:
I test in Google Chrome.

1) guess is right

```
irb(main):001:0> data = "<html><head><meta http-equiv='content-type' content='text/html; charset=utf8;'></head><body><div>hi!\xc2♥</div></body></html>"
=> "<html><head><meta http-equiv='content-type' content='text/html; charset=utf8;'></head><body><div>hi!\xC2♥</div></body></html>"

irb(main):002:0> open('test1.html', 'w') do |f|
irb(main):003:1*   f.puts(data)
irb(main):004:1> end
=> nil
```
 
![screen shot 2016-08-21 at 03 16 12](https://cloud.githubusercontent.com/assets/2876841/17833022/b3cb0dfe-674d-11e6-9f4a-d726f77b7954.png)

2) guess is not right

```
# set charset to utf16
irb(main):001:0> data = "<html><head><meta http-equiv='content-type' content='text/html; charset=utf16;'></head><body><div>hi!\xc2♥</div></body></html>"
=> "<html><head><meta http-equiv='content-type' content='text/html; charset=utf16;'></head><body><div>hi!\xC2♥</div></body></html>"

# but data is UTF-8
irb(main):005:0> data.encoding
=> #<Encoding:UTF-8>

irb(main):006:0> open('test2.html', 'w') do |f|
irb(main):007:1*   f.puts(data)
irb(main):008:1> end
=> nil
```

![screen shot 2016-08-21 at 03 22 14](https://cloud.githubusercontent.com/assets/2876841/17833075/c3b0c79e-674e-11e6-85e4-4a7eb8db30fd.png)